src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useContext, useEffect } from 'react';
+import { Filter } from '../../enums/Filter';
+
+import { SearchLink } from '../SearchLink';
+import classNames from 'classnames';
+import { PeopleContext } from '../../Context';
+
+export const PeopleFilters = () => {
+  // #region context
+
+  const {
+    activeCenturies,
+    filter,
+    sexParam,
+    query,
+    setFilter,
+    queryHandler,
+    centuriesHandler,
+    resetHandler,
+  } = useContext(PeopleContext);
+
+  // #endregion
+  // #region states
+
+  // #endregion
+  // #region values
+
+  const sexValues: (string | null)[] = [null, 'm', 'f'];
+  const filterValues = Object.values(Filter);
+
+  // #endregion
+  // #region useEffects
+
+  useEffect(() => {
+    sexValues.forEach((value, index) => {
+      if (sexParam === value) {
+        setFilter(filterValues[index]);
+      }
+    });
+  }, [sexParam]);
```

Comment:
Consider adding `setFilter` to the dependency array of this `useEffect` to ensure that the effect is re-run if `setFilter` changes. This is important for maintaining the correct behavior of the filter logic.